AliaksandrVarachai/shop-be
# Task 6

## Task 6

- Link to the integrated frontend repository: https://github.com/AliaksandrVarachai/shop-react-redux-cloudfront
- Link to a hosted web-site: https://d1fxaqepkkpfnn.cloudfront.net

### Task 6.1
- Created `catalogBatchProcess` lambda function
- Configured `serverless.yml`

### Task 6.2
- Configured an SQS queue `catalogItemsQueue`
- SQS is configured to trigger lambda `catalogBatchProcess` with `batchSize: 5`
- `catalogBatchProcess` creates all imported products in a DB

### Task 6.3
- Configured an SNS topic `createProductTopic`
- `catalogBatchProcess` sends emails via `createProductTopic` when a product created or the creation is failed

### Task 6.4
- Created a PR with made changes.

### Additional tasks
1. `catalogBatchProcess` lambda is covered by unit tests
2. SNS Filter Policy for `createProductTopic` is configured

### Test instructions with proofs
**Step 1** Upload a `CSV` file with 2 _correct_ and 2 _incorrect_ products:
```
title,description,price,count
title1,description1,10,1
title2,description2,incorrect_price,2
title3,description3,30,incorrect_count
title4,description4,40,4
```

**Step 2** Check DB. 2 _correct_ products are added:
<img width="1311" alt="Screenshot 2022-07-12 at 10 09 49" src="https://user-images.githubusercontent.com/10499432/178443973-dd1f4c1d-1a97-459f-bbb6-9b8f5047274c.png">

**Step 3** Check an email for **all messages**. 4 messages are received:
- 2 messages `Product is created`
- 2 messages `Product creation failed`
![Screenshot 2022-07-12 at 10 08 33](https://user-images.githubusercontent.com/10499432/178444701-fb261c17-1933-4b48-9015-5c280bd99ada.png)



**Step 4** Check an email for **error messages**. 2 messages are received:
- 2 messages `Product creation failed`
<img width="961" alt="Screenshot 2022-07-12 at 10 09 12" src="https://user-images.githubusercontent.com/10499432/178444976-937b39ad-4bc9-4ab4-bc25-039d777e282b.png">
